### PR TITLE
Fix auth

### PR DESCRIPTION
**Pull request by Aliics**
When we could not validate the user we would not respond with anything meaningful. This caused a lot of our crashes in the **Kinesis Analytics** apps.

### What changed:
- If we are unable to validate the access token we respond with an **AccessDeniedError**
- Rewrote the LambdaGraphQL _handleRequest_ method

### CODE REVIEW

File: src/main/java/com/fleetpin/graphql/aws/lambda/GraphQLQuery.java [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 public class GraphQLQuery {
 
-	private final String operationName;
+	private final String  operationName;
```
Comment (joshua-rule) [inline src/main/java/com/fleetpin/graphql/aws/lambda/GraphQLQuery.java:22]: what is with the extra space?

File: src/main/java/com/fleetpin/graphql/aws/lambda/LambdaGraphQL.java [hunk 2]
```diff
@@ -11,92 +11,91 @@
...
+import java.util.concurrent.ExecutionException;
 
-}
+import static com.google.common.net.HttpHeaders.AUTHORIZATION;
+
+public abstract class LambdaGraphQL<U, C extends ContextGraphQL> implements RequestHandler<APIGatewayV2ProxyRequestEvent,
+        APIGatewayV2ProxyResponseEvent> {
+    private static final Logger logger = LoggerFactory.getLogger(LambdaGraphQL.class);
+    private final ObjectMapper mapper;
+    private GraphQL build;
```
Comment (joshua-rule) [inline src/main/java/com/fleetpin/graphql/aws/lambda/LambdaGraphQL.java:35]: We should keep this immutable

File: src/test/java/com/fleetpin/graphql/aws/lambda/LambdaGraphQLTest.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        @Override
+        protected NoopGraphQLContext buildContext(final User user, final GraphQLQuery query) {
+            return noopGraphQLContext;
+        }
+    }
+
+    private static class AuthorizedGraphHandler extends LambdaGraphQL<User, NoopGraphQLContext> {
+        private final NoopGraphQLContext noopGraphQLContext;
+        private final CompletableFuture<User> validateFuture;
+        private final GraphQL graphQL;
```
Comment (joshua-rule) [inline src/test/java/com/fleetpin/graphql/aws/lambda/LambdaGraphQLTest.java:107]: This is replacing the super class build variable, I wouldn't declare it here, rather create a new protected constructor in the super class for us to call.

File: src/main/java/com/fleetpin/graphql/aws/lambda/LambdaGraphQL.java [hunk 4]
```diff
@@ -11,92 +11,91 @@
...
+            }
+
+            final var response = new APIGatewayV2ProxyResponseEvent();
+            response.setStatusCode(200);
+            response.setHeaders(Constants.GRAPHQL_RESPONSE_HEADERS);
+            response.setBody(serializedQueryResponse.toString());
+
+            return response;
+        } catch (final ExecutionException e) {
+            logger.error("Failed to validate user", e);
```
Comment (ashley-taylor) [inline src/main/java/com/fleetpin/graphql/aws/lambda/LambdaGraphQL.java:69]: You can be certain the failure was validation. Should check the cause.

I would look at adding a custom exception type for access denied so you can do an instance of check on the cause. Would probably make sense to extend one of the graphql error types as well maybe `graphql.schema.idl.errors.BaseError` or `graphql.GraphQLError`  for this exception

File: src/main/java/com/fleetpin/graphql/aws/lambda/LambdaGraphQL.java [hunk 5]
```diff
@@ -11,92 +11,91 @@
...
+
+            final var response = new ObjectNode(JsonNodeFactory.instance);
+            response.putArray(Constants.GRAPHQL_ERRORS_FIELD).add(Constants.GRAPHQL_ACCESS_DENIED);
```
Comment (ashley-taylor) [inline src/main/java/com/fleetpin/graphql/aws/lambda/LambdaGraphQL.java:72]: does not match the graphql error schema
If you do the above logic  could do something like
```
var cause = e.getCause()
if(cause instanceof GraphQLError) {
var error = (GraphQLError) = cause
var toReturn = ExecutionResultImpl.newExecutionResult().addError(error).build()

}else {
//may be a better one
var toReturn = new AbortExecutionException(cause)
}
```

File: src/main/java/com/fleetpin/graphql/aws/lambda/GraphQLQuery.java (hunk 1 above)
Comment (Aliics) [reply src/main/java/com/fleetpin/graphql/aws/lambda/GraphQLQuery.java:22]: Probably an accidental press. haha :p

File: src/main/java/com/fleetpin/graphql/aws/lambda/LambdaGraphQL.java (hunk 4 above)
Comment (Aliics) [reply src/main/java/com/fleetpin/graphql/aws/lambda/LambdaGraphQL.java:69]: Yep. Thought about this overnight haha.

File: src/test/java/com/fleetpin/graphql/aws/lambda/LambdaGraphQLTest.java (hunk 3 above)
Comment (Aliics) [reply src/test/java/com/fleetpin/graphql/aws/lambda/LambdaGraphQLTest.java:107]: Yeah, I felt iffy about the mutable field as well.

**Comment by Aliics** [comment]
Wish me luck.

**Pull request merged**